Wikipedia/Code/ArticleCollectionViewCell.swift

```diff
@@ -1,109 +1,109 @@
 import UIKit
 
 @objc(WMFArticleCollectionViewCell)
-open class ArticleCollectionViewCell: WMFExploreCollectionViewCell {
-    open var imageWidth: Int {
-        assert(false, "Subclassers must implement imageWidth")
-        return 0
-    }
+open class ArticleCollectionViewCell: CollectionViewCell {
+    let titleLabel = UILabel()
+    let descriptionLabel = UILabel()
+    let imageView = UIImageView()
+    let saveButton = SaveButton()
+    var extractLabel: UILabel?
     
-    open class var nibName: String {
-        assert(false, "Subclassers must implement nibName")
-        return "ArticleCollectionViewCell"
-    }
+    private var kvoButtonTitleContext = 0
     
-    open class var classNib: UINib {
-        return UINib(nibName: nibName, bundle: nil)
+    open override func setup() {
+        tintColor = UIColor.wmf_blueTint
+        imageView.contentMode = .scaleAspectFill
+        imageView.masksToBounds = true
+        imageView.wmf_showPlaceholder()
+        addSubview(imageView)
+        addSubview(titleLabel)
+        addSubview(descriptionLabel)
+        descriptionLabel.textColor = UIColor.wmf_customGray
+        addSubview(saveButton)
+        saveButton.tintColor = UIColor.wmf_blueTint
+        saveButton.setTitleColor(UIColor.wmf_blueTint, for: .normal)
+        saveButton.saveButtonState = .longSave
+        saveButton.addObserver(self, forKeyPath: "titleLabel.text", options: .new, context: &kvoButtonTitleContext)
+        backgroundColor = .white
+        super.setup()
```

Comment:
When we're doing frames manually, as guidance for future volunteers do we maybe want assertions somewhere verifying that we haven't set autoresizingMasks or constraints on such views?